core/src/main/java/org/infinispan/topology/ClusterCacheStatus.java

```diff
@@ -908,4 +910,73 @@ public ConsistentHash calculateConflictHash(ConsistentHash preferredHash, Set<Co
       unionHash = chf.union(unionHash, chf.rebalance(unionHash));
       return chf.updateMembers(unionHash, actualMembers, capacityFactors);
    }
+
+   @Override
+   public synchronized void queueConflictResolution(CacheTopology conflictTopology) {
+      if (resolveConflictsOnMerge()) {
+         log.debugf("Queueing conflict resolution for cache %s with members %s", cacheName, conflictTopology.getMembers());
+         queuedConflictTopology = conflictTopology;
+         startQueuedConflictResolution();
+      }
+   }
+
+   private synchronized void startQueuedConflictResolution() {
+      if (queuedConflictTopology == null)
+         return;
+
+      if (!conflictFuture.isDone()) {
+         // Check membership, if node removed then we must cancel the CR and start again below
+         // If currentTopology members size is one, cancel CR as it will be re-triggered on future merge
+         conflictFuture.cancel(true);
+      }
+
+      AdvancedCache<?, ?> cache = cacheManager.getCache(cacheName).getAdvancedCache();
+      InternalConflictManager<?, ?> conflictManager = (InternalConflictManager) ConflictManagerFactory.get(cache);
+      conflictFuture = conflictManager.resolveConflicts(queuedConflictTopology);
+      // Will execute using the StateTransferExecutor on completion
+      conflictFuture.thenRun(this::completeConflictResolution);
```

Comment:
I'd like to run `completeConflictResolution`, either to restart conflict resolution or to start a rebalance, even if `ConflictManagerFactory.get()` failed, or if `resolveConflicts` throws an exception.

If I understand correctly you don't restart CR immediately now, you only do it if there's a new graceful leave/cluster view? If so, that would leave CR in limbo if it times out, as it did in Diego's test.